product_template_pricelist_button/models/product_template.py

```diff
@@ -0,0 +1,31 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    def open_pricelist_rules(self):
+        action = super().open_pricelist_rules()
+        self.ensure_one()
+        domain = ['|','|',
+            ('product_tmpl_id', '=', self.id),
+            ('product_id', 'in', self.product_variant_ids.ids),
+            ('categ_id','<=',self.categ_id.id),
```

Comment:
Right operator is parent_of:
```suggestion
            ('categ_id','parent_of',self.categ_id.id),
```

that will retrieve "those price rules that belong to my category OR any category that is a parent of my category".

Let's see this with an example. These are some products:

![imagen](https://github.com/user-attachments/assets/2b5aea96-4d44-4057-93d2-53e9f54afd65)

We've got 24 products (green one) that belong to "Office Furniture" category, whose parent is "Saleable" and grandparent "All". Then, we find that there are 2 products whose category is "All" (the "Office Furniture" grandparent) and the other products don't belong to the desired category hierarchy.

If we select "Office Furniture" category, python shell shows that there are 24 products with such category (in my database this category has `id` 8):

![imagen](https://github.com/user-attachments/assets/78166d60-bf47-460a-8a9d-72c932185439)

If parent_of operator is used instead, Odoo will search across parent, grandparent, etc categories of desired category... and will retrieve the other 2 products (and 24+2=26!):

![imagen](https://github.com/user-attachments/assets/fefad614-efa9-48da-a544-c6952928e393)

This example was created only for direct product search, but it's basically the same for pricelist rules, you can check.

At last, `<=` operator has no sense, because we'll telling Odoo "get those pricelist rules whose category identificator is certain number or less", and category identificators are arbitrary and don't reflect anything about categories hierarchy